Support Zip64 when compressing iterables and strings

arjan-s opened the pull request:
Zip64 support currently doesn't work when the inputs for the generated zip file are iterables and/or strings. The reason for this is that __write() assumes a file size of 0 bytes for those inputs. Because of the file size the module doesn't enable Zip64 support. When it detects during compression that the file size is larger than ZIP64_LIMIT, it (correctly) raises RuntimeError('File size has increased during compressing').

This patch adds the optional (and thus backwards compatible) argument buffer_size to write_iter and writestr. This allows programs using the module to specify the buffer size that will result from the iterable or string, and in turn that allows __write() to enable Zip64 support when necessary.

Commit:
Allow passing buffer_size to write_iter and writestr

Commit:
Merge remote-tracking branch 'allanlei/master'

Commit:
New release of forked package

kageurufu committed:
Add partial flushing of ZipStreams

I use this to flush partial zips as files are streamed into
 them from requests, and then at the end add manifest and
 error files to the end of the archive

I've also added a related test and example of use

matthewatabet commented:
Would it be possible to merge this? I ran into this issue because I need to stream very large amounts of data.

arjan-s commented:
Hi @matthewatabet, it seems the original developer of this package is not active anymore. So I've forked the package as zipstream-new: https://github.com/allanlei/python-zipstream/issues/33

arjan-s committed:
Merge pull request #1 from DisruptiveLabs/partial-flush

Add partial flushing of ZipStreams

Commit:
Release 1.1.6

ryanlovett committed:
Find compression type.

arjan-s committed:
Merge pull request #2 from ryanlovett/patch-1

Find compression type.

jakepoz committed:
Stream data in order it was received

When flushing, stream out the iterators in First in first out order. Python `pop()` with no arguments would take the last path but I think it makes sense to stream the first things first.

We ran into this issue where we add a bunch of files which depend on long-running futures to provide the data. The futures hit a server which processes them roughly in order, so we get better streaming performance if we make this change.

arjan-s committed:
Merge pull request #4 from jakepoz/patch-1

Stream data in order it was received

Commit:
Release 1.1.7

arjan-s committed:
Create CHANGELOG.md

oszkarkmetti committed:
New datetime parameter into write_iter.

Commit:
Add date_time parameter to writestr() too

Commit:
Merge pull request #8 from okmetti/add_datetime_to_write_iter

Commit:
Release 1.1.8

beckedorf committed:
fix readme example

arjan-s committed:
Merge pull request #12 from beckedorf/fix-readme

fix readme example
